src/modules/ekf2/EKF/covariance.cpp

```diff
@@ -124,218 +124,144 @@ void Ekf::predictCovariance(const imuSample &imu_delayed)
 	// belong to the same group (e.g. vel_x, vel_y, vel_z)
-	const float quat_var_max = 1.0f;
-	const float vel_var_max = 1e6f;
-	const float pos_var_max = 1e6f;
-	const float gyro_bias_var_max = 1.0f;
-
-	constrainStateVar(State::quat_nominal, 0.f, quat_var_max);
-	constrainStateVar(State::vel, 1e-6f, vel_var_max);
-	constrainStateVar(State::pos, 1e-6f, pos_var_max);
-	constrainStateVar(State::gyro_bias, 0.f, gyro_bias_var_max);
-
-	// the following states are optional and are deactivated when not required
-	// by ensuring the corresponding covariance matrix values are kept at zero
-
-	// accelerometer bias states
-	if (!_accel_bias_inhibit[0] || !_accel_bias_inhibit[1] || !_accel_bias_inhibit[2]) {
-		// Find the maximum delta velocity bias state variance and request a covariance reset if any variance is below the safe minimum
-		const float minSafeStateVar = 1e-9f / sq(_dt_ekf_avg);
-		float maxStateVar = minSafeStateVar;
-		bool resetRequired = false;
-
-		for (unsigned axis = 0; axis < State::accel_bias.dof; axis++) {
-			const unsigned stateIndex = State::accel_bias.idx + axis;
-
-			if (_accel_bias_inhibit[axis]) {
-				// Skip the check for the inhibited axis
-				continue;
-			}
-
-			if (P(stateIndex, stateIndex) > maxStateVar) {
-				maxStateVar = P(stateIndex, stateIndex);
-
-			} else if (P(stateIndex, stateIndex) < minSafeStateVar) {
-				resetRequired = true;
-			}
-		}
-
-		// To ensure stability of the covariance matrix operations, the ratio of a max and min variance must
-		// not exceed 100 and the minimum variance must not fall below the target minimum
-		// Also limit variance to a maximum equivalent to a 0.1g uncertainty
-		const float minStateVarTarget = 5E-8f / sq(_dt_ekf_avg);
-		float minAllowedStateVar = fmaxf(0.01f * maxStateVar, minStateVarTarget);
 
-		for (unsigned axis = 0; axis < State::accel_bias.dof; axis++) {
-			const unsigned stateIndex = State::accel_bias.idx + axis;
-
-			if (_accel_bias_inhibit[axis]) {
-				// Skip the check for the inhibited axis
-				continue;
-			}
-
-			P(stateIndex, stateIndex) = math::constrain(P(stateIndex, stateIndex), minAllowedStateVar, sq(0.1f * CONSTANTS_ONE_G));
-		}
-
-		// If any one axis has fallen below the safe minimum, all delta velocity covariance terms must be reset to zero
-		if (resetRequired) {
-			resetAccelBiasCov();
-		}
-	}
-
-	if (force_symmetry) {
-		P.makeRowColSymmetric<State::quat_nominal.dof>(State::quat_nominal.idx);
-		P.makeRowColSymmetric<State::vel.dof>(State::vel.idx);
-		P.makeRowColSymmetric<State::pos.dof>(State::pos.idx);
-		P.makeRowColSymmetric<State::gyro_bias.dof>(State::gyro_bias.idx);
-		P.makeRowColSymmetric<State::accel_bias.dof>(State::accel_bias.idx);
-	}
+	constrainStateVar(State::quat_nominal, 1e-9f, 1.f);
+	constrainStateVar(State::vel, 1e-6f, 1e6f);
+	constrainStateVar(State::pos, 1e-6f, 1e6f);
+	constrainStateVarLimitRatio(State::gyro_bias, 1e-9f, 1.f);
+	constrainStateVarLimitRatio(State::accel_bias, 1e-9f, 1.f);
```

Comment:
review these magic numbers

```suggestion
	constrainStateVarLimitRatio(State::accel_bias, 1e-9f, 1.f);
```